src/controller/CHIPDeviceController.h

```diff
@@ -48,6 +48,38 @@ typedef void (*ErrorHandler)(ChipDeviceController * deviceController, void * app
                              const Inet::IPPacketInfo * pktInfo);
 typedef void (*MessageReceiveHandler)(ChipDeviceController * deviceController, void * appReqState, System::PacketBuffer * payload);
 
+class DLL_EXPORT PersistentStorageDelegate
```

Comment:
should we split this into a separate file?  `PersistentStorage` does not seem to really belong to DeviceController.

It would be ideal to have one class per header.